Изменила файлы ДЗ

NKVol opened the pull request:
**Изменила LionStFMS.py:**
1. Изменила реализация конечного автомата состояния льва fsm_realisation: полностью поменяла логику
2. Изменила конструктор: добавила список всех возможных состояний в конструктор: lion_list
3. Изменила objects_list: добавила объектов
4. Изменила lion_fsm_work: убрала цикл while

**Изменила LionSateFMS_Test.py:**
1. Добавила тест ввода не верного значения объекта: test_fms_bad_obj 
2. Изменила вид исключения в test_init с NameError на ValueError

NKVol committed:
Добавила ДЗ_4

NKVol committed:
Добавила .travis.yml файл

NKVol committed:
Удалила command to install dependencies

NKVol committed:
Update .travis.yml

NKVol committed:
Update LionSateFMS_Test.py

исправила ошибка в test_init

NKVol committed:
Update LionStFMS.py

исправила ошибки в коде

NKVol committed:
Update README.md

Добавила бейдж

NKVol committed:
Update README.md

исправила бейдж

NKVol committed:
Update README.md

Поправила настройки бейдж Travis ci

NKVol committed:
Update LionStFMS.py

Изменила код конструктора
Добавила метод lion_fms_work(self)

NKVol committed:
Update LionSateFMS_Test.py

Разбила тест FMS на более мелкие
Добавила проверку исходных состояний: test_init_fed, test_init_hungry
Добавила проверку работы FMS для всех состояний в цикле

NKVol committed:
Update LionSateFMS_Test.py

Добавила тест ввода не верного значения объекта: test_fms_bad_obj 
Изменила вид исключения в test_init с NameError на ValueError

NKVol committed:
Update LionStFMS.py

Изменила реализация конечного автомата состояния льва fsm_realisation: полностью поменяла логику
Изменила конструктор: добавила список всех возможных состояний в конструктор: lion_list
Изменила objects_list: добавила объектов
Изменила lion_fsm_work: убрала цикл while

NKVol committed:
Update LionStFMS.py

убрала ненужный метод и список

NKVol committed:
Update LionSateFMS_Test.py

Изменила метод test_fms

NKVol committed:
Update LionStFMS.py

Убрала obj_list

skyline-gleb commented on LionSateFMS_Test.py:
```diff
@@ -0,0 +1,79 @@
+        lion.fsm_realisation("дерево")
+        lion.fsm_realisation("дерево")
+        lion.fsm_realisation("охотник")
+        lion.fsm_realisation("антилопа")
```
почему это тест? не похож, совсем)

NKVol replied:
Захотела проверить как система работает, если ей на вход подряд подавать разные значения)
Не нужно так делать?
Как в целом реализация?

skyline-gleb replied:
что такое тест? думаю когда ответишь на вопрос, поймешь что тут не так.
в целом я тесты не смотрел))

NKVol committed:
Update LionStFMS.py

Изменения:
1.Убрала комментарии
2.Изменила список на словарь в конструкторе
3.Изменила логику FMS

NKVol committed:
Update LionSateFMS_Test.py

Изменения:
1.Убрала комментарии
2.Убрала последний не нужный тест test_fms

NKVol commented:
**Изменила LionStFMS.py:**
1.Убрала комментарии
2.Изменила список на словарь в конструкторе
3.Изменила логику FMS

**Изменила LionSateFMS_Test.py:**
1.Убрала комментарии
2.Убрала последний не нужный тест test_fms

skyline-gleb commented on LionStFMS.py:
```diff
@@ -0,0 +1,33 @@
+            action = self.state_list[self.state][obj][0]
+            state = self.state_list[self.state][obj][1]
+        except Exception:
+            raise ValueError("Вы ввели не верный объект! Нет такой комбинации состояние + объект")
```
весь этот try/except фу-фу-фу. у тебя есть всё для того чтобы проверить корректный ли `obj` и сказать, что для конкретного текущего состояния `state` не определено перехода с заданным `obj`
а подсказка в этот раз будет такая - у словаря есть пара клевых функций, возвращающих, например, список всех значений.

skyline-gleb commented on LionStFMS.py:
```diff
@@ -0,0 +1,33 @@
+class LionStateFSM:
+
+    def __init__(self, state):
+        if state == "сытый" or state == "голодный":
```
и сразу обращаю внимание на эту строку кода....теперь она ох как хочет измениться и стать гораздо лучше
